scripts/performance/run-performace-tests-functions.sh

```diff
@@ -0,0 +1,173 @@
+   echo -e "\nCreating '$FILE1'"
+   dd if=/dev/zero of=$FILE1 bs=1M count=10
+   echo -e "\nCreating '$FILE2'"
+   date > $FILE2
+
+   add_common_separator "Uploading '$FILE1' file to '$BUCKET' bucket"
+   aws s3 cp $FILE1 s3://$BUCKET/file10MB
+   check_status "Failed to upload '$FILE1' to '$BUCKET'"
+   add_common_separator "Uploading '$FILE2' file to '$BUCKET' bucket"
+   aws s3 cp $FILE2 s3://$BUCKET
+   check_status "Failed to upload '$FILE2' to '$BUCKET'"
+
+   add_common_separator "List files in '$BUCKET' bucket"
+   aws s3 ls s3://$BUCKET
+   check_status "Failed to list files in '$BUCKET'"
+
+   add_common_separator "Download '$FILE1' as 'file10mbDn' and check diff"
+   aws s3 cp s3://$BUCKET/file10MB file10mbDn
+   check_status "Failed to download '$FILE1' as 'file10mbDn' from '$BUCKET'"
+   FILE_DIFF=$(diff $FILE1 file10mbDn)
+
+   if [[ $FILE_DIFF ]]; then
+      echo -e "\nDIFF Status: $FILE_DIFF"
+   else
+      echo -e "\nDIFF Status: The files $FILE1 and file10mbDn are similar."
+   fi
+
+   add_common_separator "Remove all files in '$BUCKET' bucket"
+   aws s3 rm s3://$BUCKET --recursive
+   check_status "Failed to delete all files from '$BUCKET'"
+
+   add_common_separator "Remove '$BUCKET' bucket"
+   aws s3 rb s3://$BUCKET
+   check_status "Failed to delete '$BUCKET'"
+   add_primary_separator "\tSuccessfully Passed IO Sanity Testing"
+}
+
+function clone_segate_tools_repo() {
+    if [ -z "$SCRIPT_LOCATION" ]; then echo "SCRIPT_LOCATION not provided.Exiting..."; exit 1; fi
+    if [ -z "$GITHUB_TOKEN" ]; then echo "GITHUB_TOKEN not provided.Exiting..."; exit 1; fi
+    if [ -z "$CORTX_TOOLS_REPO" ]; then echo "CORTX_TOOLS_REPO not provided.Exiting..."; exit 1; fi
+    if [ -z "$CORTX_TOOLS_BRANCH" ]; then echo "CORTX_TOOLS_BRANCH not provided.Exiting..."; exit 1; fi
+
+    rm -rf $SCRIPT_LOCATION
+    yum install git -y
+    git clone https://$GITHUB_TOKEN@github.com/$CORTX_TOOLS_REPO $SCRIPT_LOCATION
+    pushd $SCRIPT_LOCATION
+    git checkout $CORTX_TOOLS_BRANCH
+    popd
+}
+
+function update_setup_confiuration() {
+    #Update root password in config.yaml
+    sed -i '/CLUSTER_PASS/s/seagate1/'$PRIMARY_CRED'/g' $SCRIPT_LOCATION/performance/PerfPro/roles/benchmark/vars/config.yml
+    sed -i -e '/NODES/{n;s/.*/  - 1: '$PRIMARY_NODE'/}' -e '/CLIENTS/{n;s/.*/  - 1: '$CLIENT_NODE'/}' $SCRIPT_LOCATION/performance/PerfPro/roles/benchmark/vars/config.yml
+    sed -i '/BUILD_URL/s/\:/: '${BUILD_URL//\//\\/}'/g' $SCRIPT_LOCATION/performance/PerfPro/roles/benchmark/vars/config.yml
+}
+
+function execute_perfpro() {
+    yum install ansible -y
+    pushd $SCRIPT_LOCATION/performance/PerfPro
+    ansible-playbook perfpro.yml -i inventories/hosts --extra-vars '{ "EXECUTION_TYPE" : "sanity" ,"REPOSITORY":{"motr":"cortx-motr","rgw":"cortx-rgw"} , "COMMIT_ID": { "main" : "d1234c" , "dev" : "a5678b"},"PR_ID" : "cortx-rgw/1234" , "USER":"Shailesh Vaidya","GID" : "729494" }' -v
```

Comment:
are these vars going to be hardcoded?

Reply:
Yes for now. Need to work with performance team further to avoid GID and other details.